ddtrace/contrib/pytest/plugin.py

```diff
@@ -161,9 +165,10 @@ def pytest_runtest_makereport(item, call):
     if span is None:
         return
 
-    called_without_status = call.when == "call" and span.get_tag(test.STATUS) is None
-    failed_setup = call.when == "setup" and call.excinfo is not None
-    if not called_without_status and not failed_setup:
```

Comment:
I _think_ the logic was unnecessarily complex. What we want to do is avoid overwriting the status of a test if it fails but the teardown is successful.

Reply:
does this still pass the expected tests? are there any tests missing?

Reply:
yeah I'd say at least a test is missing. On it

Reply:
added 3 tests: for internal exceptions, and for errors in setup/teardown in https://github.com/DataDog/dd-trace-py/pull/2836/commits/0dc0a62806b7427f624b5ddd01c6f275d15a1d47